python/tests/test_nn.py

```diff
@@ -792,6 +792,24 @@ def test_log_cosh_loss(self):
         loss = nn.losses.log_cosh_loss(inputs, targets, reduction="mean")
         self.assertAlmostEqual(loss.item(), 0.433781, places=6)
 
+    def test_dropout(self):
+        x = mx.ones((2, 4))
+        y = nn.Dropout(0.5)(x)
+        self.assertTrue(y.shape, x.shape)
+        self.assertTrue(y.dtype, mx.float32)
```

Comment:
Could you add a test where `x` is in `bfloat16` or `float16` as well (here and below?)

Reply:
Done!